### PR TITLE
Warning fixes

### PR DESCRIPTION
**Pull request by bluescarni**
@ukoethe This is the initial batch of compiler warning fixes. I have not included the ``SYSTEM`` header changes (as you mentioned you would work on it yourself).

### CODE REVIEW

File: include/vigra/accumulator.hxx [hunk 1]
```diff
@@ -710,7 +710,7 @@ template <>
 struct CollectAccumulatorNames<void>
 {
     template <class BackInsertable>
-    static void exec(BackInsertable & a, bool skipInternals=true)
+    static void exec(BackInsertable &, bool = true)
```
Comment (ukoethe) [inline include/vigra/accumulator.hxx]: I'd prefer
```
    static void exec(BackInsertable & a, bool /* skipInternals */ = true)
```

File: include/vigra/blockify.hxx [hunk 2]
```diff
@@ -84,7 +84,7 @@ struct blockify_impl<0>
                      Shape current_block_begin,
                      Shape current_block_end,
                      Shape current_block_pos,
-                     Shape block_shape)
+                     Shape)
```
Comment (ukoethe) [inline include/vigra/blockify.hxx]: `/* block_shape */`

File: include/vigra/blockwise_labeling.hxx [hunk 3]
```diff
@@ -172,7 +172,7 @@ blockwiseLabeling(DataBlocksIterator data_blocks_begin, DataBlocksIterator data_
         //std::iota(ids.begin(), ids.end(), 0 );
 
         parallel_foreach(options.getNumThreads(), d,
-            [&](const int threadId, const uint64_t i){
+            [&](const int, const uint64_t i){
```
Comment (ukoethe) [inline include/vigra/blockwise_labeling.hxx]: `/* threadId */`

File: include/vigra/blockwise_watersheds.hxx [hunk 4]
```diff
@@ -61,6 +62,7 @@ void prepareBlockwiseWatersheds(const Overlaps<DataArray>& overlaps,
                                 BlockwiseLabelOptions const & options)
 {
     static const unsigned int N = DataArray::actual_dimension;
+    ignore(N);
```
Comment (ukoethe) [inline include/vigra/blockwise_watersheds.hxx]: rename `ignore_argument()`

File: include/vigra/blockwise_labeling.hxx [hunk 5]
```diff
@@ -137,7 +137,7 @@ template <class DataBlocksIterator, class LabelBlocksIterator,
           class Equal, class Mapping>
 typename BlockwiseLabelingResult<LabelBlocksIterator>::type
 blockwiseLabeling(DataBlocksIterator data_blocks_begin, DataBlocksIterator data_blocks_end,
-                  LabelBlocksIterator label_blocks_begin, LabelBlocksIterator label_blocks_end,
+                  LabelBlocksIterator label_blocks_begin, LabelBlocksIterator,
```
Comment (ukoethe) [inline include/vigra/blockwise_labeling.hxx]: It the two ranges must have the same length, don't ignore the end iterator but add a precondition instead.

File: include/vigra/blockwise_watersheds.hxx [hunk 6]
```diff
@@ -72,7 +74,7 @@ void prepareBlockwiseWatersheds(const Overlaps<DataArray>& overlaps,
 
     parallel_foreach(options.getNumThreads(),
         itBegin,end,
-        [&](const int threadId, const Coordinate  iterVal){
+        [&](const int, const Coordinate  iterVal){
```
Comment (ukoethe) [inline include/vigra/blockwise_watersheds.hxx]: `/* threadId */`

File: include/vigra/graph_algorithms.hxx [hunk 7]
```diff
@@ -738,7 +737,7 @@ namespace vigra{
 
     struct RawPriorityFunctor{
         template<class L, class T>
-        T operator()(const L label,const T  priority)const{
+        T operator()(const L,const T  priority)const{
```
Comment (ukoethe) [inline include/vigra/graph_algorithms.hxx]: call the type `LabelType`

File: include/vigra/graph_item_impl.hxx [hunk 8]
```diff
@@ -43,9 +43,9 @@ namespace vigra{
             typedef typename GRAPH::NodeStorage::AdjacencyElement AdjacencyElement;
 
             static bool valid(
-                const GRAPH & g,
-                const AdjacencyElement & adj,
-                const typename GRAPH::index_type ownNodeId
+                const GRAPH &,
```
Comment (ukoethe) [inline include/vigra/graph_item_impl.hxx]: `/* ownNodeId */`

File: include/vigra/graph_item_impl.hxx [hunk 9]
```diff
@@ -43,9 +43,9 @@ namespace vigra{
...
+                const AdjacencyElement &,
+                const typename GRAPH::index_type
```
Comment (ukoethe) [inline include/vigra/graph_item_impl.hxx]: `/* ownNodeId */`

File: include/vigra/graph_item_impl.hxx [hunk 10]
```diff
@@ -54,7 +54,7 @@ namespace vigra{
              static ResultType transform(
                 const GRAPH & g,
                 const AdjacencyElement & adj,
-                const typename GRAPH::index_type ownNodeId
+                const typename GRAPH::index_type
```
Comment (ukoethe) [inline include/vigra/graph_item_impl.hxx]: `/* ownNodeId */`

File: include/vigra/graph_item_impl.hxx [hunk 11]
```diff
@@ -68,9 +68,9 @@ namespace vigra{
             typedef typename GRAPH::NodeStorage::AdjacencyElement AdjacencyElement;
 
             static bool valid(
-                const GRAPH & g,
-                const AdjacencyElement & adj,
-                const typename GRAPH::index_type ownNodeId
+                const GRAPH &,
+                const AdjacencyElement &,
+                const typename GRAPH::index_type
```
Comment (ukoethe) [inline include/vigra/graph_item_impl.hxx]: `/* ownNodeId */`

File: include/vigra/graph_item_impl.hxx [hunk 12]
```diff
@@ -78,7 +78,7 @@ namespace vigra{
             static ResultType transform(
                 const GRAPH & g,
                 const AdjacencyElement & adj,
-                const typename GRAPH::index_type ownNodeId
+                const typename GRAPH::index_type
```
Comment (ukoethe) [inline include/vigra/graph_item_impl.hxx]: `/* ownNodeId */`

File: include/vigra/graph_item_impl.hxx [hunk 13]
```diff
@@ -102,7 +102,7 @@ namespace vigra{
...
```
Comment (ukoethe) [inline include/vigra/graph_item_impl.hxx]: `/* ownNodeId */`

File: include/vigra/graph_item_impl.hxx [hunk 14]
```diff
@@ -137,9 +137,9 @@ namespace vigra{
             typedef typename GRAPH::NodeStorage::AdjacencyElement AdjacencyElement;
             
             static bool valid(
-                const GRAPH & g,
-                const AdjacencyElement & adj,
-                const typename GRAPH::index_type ownNodeId
+                const GRAPH &,
+                const AdjacencyElement &,
+                const typename GRAPH::index_type
```
Comment (ukoethe) [inline include/vigra/graph_item_impl.hxx]: `/* ownNodeId */`

File: include/vigra/graph_item_impl.hxx [hunk 15]
```diff
@@ -162,16 +162,16 @@ namespace vigra{
...
```
Comment (ukoethe) [inline include/vigra/graph_item_impl.hxx]: `/* ownNodeId */`

File: include/vigra/graph_item_impl.hxx [hunk 16]
```diff
@@ -162,16 +162,16 @@ namespace vigra{
...
             ){
                 return  true;
             } 
             ResultType static transform(
                 const GRAPH & g,
                 const AdjacencyElement & adj,
-                const typename GRAPH::index_type ownNodeId
+                const typename GRAPH::index_type
```
Comment (ukoethe) [inline include/vigra/graph_item_impl.hxx]: `/* ownNodeId */`

File: include/vigra/hdf5impex.hxx [hunk 17]
```diff
@@ -3353,7 +3353,7 @@ inline void readHDF5(const HDF5ImportInfo &info, MultiArrayView<N, T, StrideTag>
 }
 
 template<unsigned int N, class T, class StrideTag>
-void readHDF5(const HDF5ImportInfo &info, MultiArrayView<N, T, StrideTag> array, const hid_t datatype, const int numBandsOfType)
+void readHDF5(const HDF5ImportInfo &info, MultiArrayView<N, T, StrideTag> array, const hid_t, const int)
```
Comment (ukoethe) [inline include/vigra/hdf5impex.hxx]: suspicious - why are the parameters unused?

File: include/vigra/hdf5impex.hxx [hunk 18]
```diff
@@ -3568,7 +3568,7 @@ inline void writeHDF5(const char* filePath, const char* pathInFile, const MultiA
 }
 
 template<unsigned int N, class T, class StrideTag>
-void writeHDF5(const char* filePath, const char* pathInFile, const MultiArrayView<N, T, StrideTag> & array, const hid_t datatype, const int numBandsOfType)
+void writeHDF5(const char* filePath, const char* pathInFile, const MultiArrayView<N, T, StrideTag> & array, const hid_t, const int)
```
Comment (ukoethe) [inline include/vigra/hdf5impex.hxx]: suspicious - why are the parameters unused?

**Comment by bluescarni** [comment]
@ukoethe I have made the changes we discussed yesterday, after merging in the RC branch. Could you please take a look at those HDF5 function? I removed them and everything seems to compile & test without errors, but my understanding yesterday was that the write function was being used in the bindings.

The only remaining warning is in ``slic.hxx`` (whose code you mentioned you would rework eventually - in the meantime I did remove the pragma statements).

**Pull request merged**